src/database/sql_database.cc

```diff
@@ -855,12 +851,9 @@ std::vector<std::shared_ptr<CdsObject>> SQLDatabase::browse(const std::unique_pt
     commit("browse");
 
     std::vector<std::shared_ptr<CdsObject>> result;
-
-    std::unique_ptr<SQLRow> row;
-    while ((row = sqlResult->nextRow())) {
-        auto obj = createObjectFromRow(row);
-        result.push_back(obj);
-        row = nullptr; // clear out content from unique_ptr
+    result.reserve(sqlResult->getNumRows());
+    while (auto row = sqlResult->nextRow()) {
```

Comment:
For readability we prefer not to mix assignments with checks and leave optimization to the compiler.

Reply:
It's now fixed wherever suitable